app/services/distribution_destroy_service.rb

```diff
@@ -1,33 +1,12 @@
-class DistributionDestroyService
-  attr_reader :distribution, :error
-
+class DistributionDestroyService < DistributionService
   def initialize(distribution_id)
     @distribution_id = distribution_id
   end
 
   def call
-    @distribution = Distribution.find(@distribution_id)
-    @organization = distribution.organization
-
-    @distribution.transaction do
+    perform_distribution_service do
       @distribution.destroy!
       @distribution.storage_location.increase_inventory(@distribution)
```

Comment:
The method you wrote in the superclass looks like it will make this safer:

```suggestion
      distribution.destroy!
      distribution.storage_location.increase_inventory(distribution)
```